File: app/src/main/java/com/niusounds/matomeviewer/data/Article.java
package com.niusounds.matomeviewer.data;

import java.io.Serializable;
import java.util.Date;

public class Article implements Serializable, Comparable<Article> {
    public String link;
    public String title;
    public String imageUrl;
    public String description;
    public Date date;
    public String pubDate;
    public String contentEncoded;

    @Override
    public int compareTo(Article article) {
        if (article == null || article.date == null || date == null) return 0;
        return article.date.compareTo(date);
    }
}

File: app/src/main/java/com/niusounds/matomeviewer/data/Article.kt
package com.niusounds.matomeviewer.data

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity
class Article : Serializable, Comparable<Article> {
    @PrimaryKey
    var link: String = ""
    var title: String? = null
    var imageUrl: String? = null
    var description: String? = null
    @Ignore
    var date: Date? = null
    var pubDate: String? = null
    var contentEncoded: String? = null

    override fun compareTo(other: Article): Int =
            if (other.date == null || date == null) 0
            else other.date!!.compareTo(date)
}
